sim/build.sbt

```diff
@@ -4,16 +4,60 @@ import Tests._
 val apiDirectory = settingKey[String]("The site directory into which the published scaladoc should placed.")
 apiDirectory := "latest"
 
+def filenames(tempDir: File, fs: Seq[File]): Seq[String] =
+for(f <- fs) yield {
+  sbtassembly.AssemblyUtils.sourceOfFileForMerge(tempDir, f) match {
+    case (path, base, subDirPath, false) => subDirPath
+    case (jar, base, subJarPath, true) => jar + ":" + subJarPath
+  }
+}
+
+// custom sbtassemblly.MergeStrategy
+// removes merge candidates that have 'rocketchip' in the name of their JAR
+// hopefully leaving only a single candidate to be chosen as the member of 
+// our assembly JAR.  If there's still multiple candidates, error.
+val notRocketMergeStrategy = new sbtassembly.MergeStrategy {
+  val name = "notRocket"
+  def apply(tempDir: File, path: String, files: Seq[File]) = {
+    val filtered = files collect { f =>
+      sbtassembly.AssemblyUtils.sourceOfFileForMerge(tempDir, f) match {
+        case (jar, _, _, true) if !jar.toString.contains("rocketchip") => f
+      }
+    }
+    if (filtered.size == 1) Right(Seq(filtered.head -> path))
+    else Left("still have multiple files after removing rocketchip for same target path:" +
+      filenames(tempDir, filtered).mkString("\n", "\n", "")
+      )
+  }
+}
+
+// custom sbtassemblly.MergeStrategy
+// keeps merge candidates that have 'rocketchip' in the name of their JAR
+// hopefully leaving only a single candidate to be chosen as the member of 
+// our assembly JAR.  If there's still multiple candidates, error.
+val useRocketMergeStrategy = new sbtassembly.MergeStrategy {
+  val name = "useRocket"
+  def apply(tempDir: File, path: String, files: Seq[File]) = {
+    val filtered = files collect { f =>
+      sbtassembly.AssemblyUtils.sourceOfFileForMerge(tempDir, f) match {
+        case (jar, _, _, true) if jar.toString.contains("rocketchip") => f
+      }
+    }
+    if (filtered.size == 1) Right(Seq(filtered.head -> path))
+    else Left("multiple candidates have rocketchip in their jar name for target:" +
+      filenames(tempDir, filtered).mkString("\n", "\n", "")
+      )
+  }
+}
```

Comment:
If @jackkoenig or anyone else knows of a more Scalatastic way to write this so that the code doesn't have to be duplicated, I'd be all for it.  I was thinking that I potentially need to create a new class that extends `sbtassembly.MergeStrategy` with another Bool parameter that is used to control the polarity and select the name and error string.

Maybe I'm trying too hard though.